### PR TITLE
Add support for component creation 

### PR DESCRIPTION
**Pull request by Revolyssup**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Lint issue seems to be persistent

**Comment by Revolyssup** [comment]
> Lint issue seems to be persistent

Gosec is complaining because I am taking write permissions while creating components. Will have to ignore it in the workflow

**Comment by Revolyssup** [comment]
> Lint issue seems to be persistent

green

File: adapter/oam.go [hunk 1]
```diff
@@ -163,6 +165,92 @@ func (or *OAMRegistrant) Register() error {
...
+		err := os.Remove(path) //Remove the existing file, before overriding it
+		if err != nil {
+			return err
+		}
+	}
+	return ioutil.WriteFile(path, data, 0777)
+}
+
+//getNameFromWorkloadDefinition takes out name from workload definition
+func getNameFromWorkloadDefinition(definition []byte) string {
```
Comment (leecalcote) [inline adapter/oam.go:244]: Is this name being run through string manipulation for display formatting?

File: adapter/oam.go [hunk 2]
```diff
@@ -163,6 +165,92 @@ func (or *OAMRegistrant) Register() error {
...
+	case HelmCHARTS:
+		comp, err = manifests.GetFromHelm(scfg.URL, manifests.SERVICE_MESH, scfg.Config)
+	default:
+		return ErrCreatingComponents(errors.New("invalid generation method. Must be either Manifests or HelmCharts"))
+	}
+	if err != nil {
+		return ErrCreatingComponents(err)
+	}
+	if comp == nil {
+		return ErrCreatingComponents(errors.New("nil components"))
```
Comment (leecalcote) [inline adapter/oam.go:204]:
```suggestion
		return ErrCreatingComponents(errors.New("no components found"))
```

File: adapter/oam.go (hunk 1 above)
Comment (Revolyssup) [reply adapter/oam.go:244]: No it's not.

File: adapter/oam.go (hunk 2 above)
Comment (Revolyssup) [reply adapter/oam.go:204]: okay

File: adapter/error.go [hunk 3]
```diff
@@ -135,6 +135,12 @@ func ErrOAMRetry(err error) error {
...
 }
 
+//will be depracated
 func ErrGenerateComponents(err error) error {
 	return errors.New(ErrGenerateComponentsCode, errors.Alert, []string{"error generating components"}, []string{err.Error()}, []string{"Invalid component generation method passed, Some invalid field passed in DynamicComponentsConfig"}, []string{"Pass the correct GenerationMethod in DynamicComponentsConfig", "Pass the correct fields in DynamicComponentsConfig"})
 }
+
+//ErrCreatingComponents
+func ErrCreatingComponents(err error) error {
+	return errors.New(ErrCreatingComponentsCode, errors.Alert, []string{"error creating components"}, []string{err.Error()}, []string{"Invalid Path or version passed in static configuration"}, []string{"Make sure to pass correct configuration"})
```
Comment (leecalcote) [inline adapter/error.go:145]: Think through with likely scenarios here are and try to provide guidance for our future selves. One more common issue could be that the location of a given service mesh's  helm charts have moved location and that location should be verified. Will you include this and other possible probable causes and suggested mediations?

File: adapter/oam.go (hunk 1 above)
Comment (leecalcote) [reply adapter/oam.go:244]: Maybe this isn't an appropriate location for such manipulation. Where is the central set of string manipulation functions? Are they referenced by this static component generation functions?

Comment (Revolyssup) [reply adapter/oam.go:244]: I am not manipulating anything here. I am just extracting the names to set the names of the files.

File: adapter/error.go (hunk 3 above)
Comment (Revolyssup) [reply adapter/error.go]: Made changes

File: adapter/oam.go (hunk 1 above)
Comment (leecalcote) [reply adapter/oam.go:244]: Where is the single location in which string manipulation logic is kept?

File: adapter/error.go (hunk 3 above)
Comment (leecalcote) [reply adapter/error.go]: Thank you.

File: adapter/oam.go (hunk 1 above)
Comment (leecalcote) [reply adapter/oam.go:244]: We can follow up this conversation post-merge. 👍

**Comment by leecalcote** [comment]
How is Meshery Server using these same functions for registration of Kubernetes components?

**Comment by Revolyssup** [comment]
> How is Meshery Server using these same functions for registration of Kubernetes components?

Meshery server doesn't use these functions. Meshery server uses the function in meshkit

**Comment by leecalcote** [comment]
@Revolyssup are we duplicating these functions, then? If not, what warrants use of different functions for component registration? Why does Meshery Server's approach to component generation and that of Meshery Adapters need to be any different?

File: adapter/oam.go (hunk 1 above)
Comment (leecalcote) [reply adapter/oam.go:244]: Merged. Address the question above when you have time.

Comment (Revolyssup) [reply adapter/oam.go:244]: String manipulation is also present in meshkit function

**Comment by Revolyssup** [comment]
> 

To summarize:
1. Meshkit has the function which actually generates components. This function is used everywhere- in adapters as well as server.
2. Meshery server directly uses the meshkit functions. Generates those components and saves them in the database. Since meshery server owns the database, there is no such thing as "registeration" so we can directly use the meshkit function.
3. Adapters have two steps, creation and registeration. Since all adapters have these two steps, it made sense to create a thin abstraction in meshery adapter library. So in adapters, instead of directly calling meshkit functions, we call the functions in adapter library which cater specific to adapters. The function in adapter library calls in the meshkit function in turn.

**Pull request merged**